File: java/01-commands/tests/sourcecode/src/test/java/com/training/app/MultiplesOf3or5/MultiplesOf3or5TestTemplate2.java
package com.training.app.MultiplesOf3or5;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class MultiplesOf3or5TestTemplate2 {

    @Test
    public void givenInput_ReturnExpectedSum() {
        int[][] testData = {
            {-1, 0},
            {0, 0},
            {4, 3},
            {10, 23},
            {20, 78},
            {200, 9168}
        };

        for (int[] data : testData) {
            int input = data[0];
            int expectedSum = data[1];
            MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(input);
            assertEquals(expectedSum, multiplesOf3or5.sum());
        }
    }
}

File: dotnet/01-commands/tests/MultiplesOf3or5/MultiplesOf3or5TestTemplate2.cs
namespace MultiplesOf3or5
{
    public class MultiplesOf3or5TestTemplate1
    {
        [Theory]
        [InlineData(-1, 0)]
        [InlineData(0, 0)]
        [InlineData(4, 3)]
        [InlineData(10, 23)]
        [InlineData(20, 78)]
        [InlineData(200, 9168)]
        public void GivenInput_ReturnExpectedSum(int input, int expectedSum)
        {
            var multiplesOf3or5 = new MultiplesOf3or5(input);
            Assert.Equal(expectedSum, multiplesOf3or5.Sum());
        }
    }
}
